src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -259,6 +254,16 @@ private static void addBodyIfExists(HttpURLConnection connection, Request<?> req
             throws IOException, AuthFailureError {
         byte[] body = request.getBody();
         if (body != null) {
+            addBodyIfExists(connection, request, body);
+        }
+    }
+
+    private static void addBodyIfExists(HttpURLConnection connection, Request<?> request, byte[] body)
+            throws IOException, AuthFailureError {
+        if (body != null) {
```

Comment:
both the callers already check that body != null, so either remove this check here and rename this method "addBody" (my preference), or remove the checks above.